node/core/pvf/src/execute/queue.rs

```diff
@@ -175,24 +180,35 @@ fn handle_to_queue(queue: &mut Queue, to_queue: ToQueue) {
 
 async fn handle_mux(queue: &mut Queue, event: QueueEvent) {
 	match event {
-		QueueEvent::Spawn((idle, handle)) => {
-			queue.workers.spawn_inflight -= 1;
-
-			let worker = queue.workers.running.insert(WorkerData { idle: Some(idle), handle });
-
-			if let Some(job) = queue.queue.pop_front() {
-				assign(queue, worker, job);
-			}
+		QueueEvent::Spawn(idle, handle) => {
+			handle_worker_spawned(queue, idle, handle);
 		},
-		QueueEvent::StartWork(worker, outcome, result_tx) => {
-			handle_job_finish(queue, worker, outcome, result_tx);
+		QueueEvent::StartWork(worker, outcome, artifact_id, result_tx) => {
+			handle_job_finish(queue, worker, outcome, artifact_id, result_tx);
 		},
 	}
 }
 
+fn handle_worker_spawned(queue: &mut Queue, idle: IdleWorker, handle: WorkerHandle) {
+	queue.workers.spawn_inflight -= 1;
+	let worker = queue.workers.running.insert(WorkerData { idle: Some(idle), handle });
+
+	tracing::debug!(target: LOG_TARGET, "worker spawned {:?}", worker);
```

Comment:
```suggestion
	tracing::debug!(target: LOG_TARGET, ?worker, "worker spawned");
```

Reply:
Just fwiw my predicate on whether to make it a field or part of the formatted string was wether I would personally look up on this value or not. I guess it doesn't really make much sense